src/charm.py

```diff
@@ -0,0 +1,415 @@
+                "summary": "routectl layer",
+                "description": "pebble config layer for routectl",
+                "services": {
+                    self._routectl_service_name: {
+                        "override": "replace",
+                        "startup": "enabled",
+                        "command": "/opt/bess/bessctl/conf/route_control.py -i access core",
+                        "environment": self._routectl_environment_variables,
+                    },
+                },
+            }
+        )
+
+    @property
+    def _web_pebble_layer(self) -> Layer:
+        """Returns pebble layer for the web container.
+
+        Returns:
+            Layer: Pebble Layer
+        """
+        return Layer(
+            {
+                "summary": "web layer",
+                "description": "pebble config layer for web",
+                "services": {
+                    self._web_service_name: {
+                        "override": "replace",
+                        "startup": "enabled",
+                        "command": "bessctl http 0.0.0.0 8000",
+                    },
+                },
+            }
+        )
+
+    @property
+    def _pfcp_agent_pebble_layer(self) -> Layer:
+        """Returns pebble layer for the pfcp agent container.
+
+        Returns:
+            Layer: Pebble Layer
+        """
+        return Layer(
+            {
+                "summary": "pfcp agent layer",
+                "description": "pebble config layer for pfcp agent",
+                "services": {
+                    self._pfcp_agent_service_name: {
+                        "override": "replace",
+                        "startup": "enabled",
+                        "command": f"pfcpiface -config {PFCP_AGENT_CONTAINER_CONFIG_PATH}/{CONFIG_FILE_NAME}",  # noqa: E501
+                    },
+                },
+            }
+        )
+
+    @property
+    def _bessd_environment_variables(self) -> dict:
+        """Returns environment variables for the bessd service.
+
+        Returns:
+            dict: Environment variables
+        """
+        return {
+            "CONF_FILE": f"{BESSD_CONTAINER_CONFIG_PATH}/{CONFIG_FILE_NAME}",
+        }
+
+    @property
+    def _routectl_environment_variables(self) -> dict:
+        """Returns environment variables for the routectl service.
+
+        Returns:
+            dict: Environment variables
+        """
+        return {
+            "PYTHONUNBUFFERED": "1",
+        }
+
+    @property
+    def core_network_gateway(self) -> Optional[str]:
+        """Core network gateway IP address."""
+        return self.model.config.get("core-gateway")
+
+    @property
+    def access_network_gateway(self) -> Optional[str]:
+        """Access network gateway IP address."""
+        return self.model.config.get("access-gateway")
+
+    @property
+    def gnb_subnet(self) -> Optional[str]:
+        """Gnodeb subnet."""
+        return self.model.config.get("gnb-subnet")
```

Comment:
Why do we need these props to be public?

Reply:
Fixed